runtime/src/mm/cpp/Memory.cpp

```diff
@@ -75,3 +93,57 @@ extern "C" RUNTIME_NOTHROW void ClearTLS(MemoryState* memory) {
 extern "C" RUNTIME_NOTHROW ObjHeader** LookupTLS(void** key, int index) {
     return mm::ThreadRegistry::Instance().CurrentThreadData()->tls().Lookup(key, index);
 }
+
+extern "C" RUNTIME_NOTHROW void* CreateStablePointer(ObjHeader* object) {
+    auto* threadData = mm::ThreadRegistry::Instance().CurrentThreadData();
+    return mm::StableRefRegistry::Instance().RegisterStableRef(threadData, object);
+}
+
+extern "C" RUNTIME_NOTHROW void DisposeStablePointer(void* pointer) {
+    auto* threadData = mm::ThreadRegistry::Instance().CurrentThreadData();
+    auto* node = static_cast<mm::StableRefRegistry::Node*>(pointer);
+    mm::StableRefRegistry::Instance().UnregisterStableRef(threadData, node);
+}
+
+extern "C" RUNTIME_NOTHROW OBJ_GETTER(DerefStablePointer, void* pointer) {
+    auto* node = static_cast<mm::StableRefRegistry::Node*>(pointer);
+    ObjHeader* object = **node;
+    RETURN_OBJ(object);
+}
+
+extern "C" RUNTIME_NOTHROW OBJ_GETTER(AdoptStablePointer, void* pointer) {
+    auto* threadData = mm::ThreadRegistry::Instance().CurrentThreadData();
+    auto* node = static_cast<mm::StableRefRegistry::Node*>(pointer);
+    ObjHeader* object = **node;
+    mm::StableRefRegistry::Instance().UnregisterStableRef(threadData, node);
+    RETURN_OBJ(object);
+}
+
+extern "C" ForeignRefContext InitLocalForeignRef(ObjHeader* object) {
```

Comment:
It should probably return `nullptr` instead.
IIRC, this function is generally a hack to make a "foreign ref" that has owner but don't actually retain anything, i.e. not a strong reference.
Since we won't need to restrict ownership, returning `nullptr` should be ok.
If unsure, please keep it as stub instead for now.

Reply:
I'm indeed unsure. Will keep it as it is then.

Reply:
Keeping it as is might be harmful, it might lead to memory leaks later.
Let's make it a `RuntimeCheck`-stub again.

Reply:
Ok, done.